src/dso/_mv.py

```diff
@@ -0,0 +1,308 @@
+    log.debug(f"Absolute target base directory: {target_absolute_dir}")
+
+    if not source_absolute_path.exists():
+        log.error(f"[red]{source} does not exist!")
+        exit(1)
+
+    # Currently is not allowed to move the folder or stage to a directory
+    # which does not exist.
+    if not target_absolute_dir.exists():
+        log.error(f"[red]{target_relative_dir} does not exist. Target base directory must already exist.")
+        exit(1)
+
+    if target_absolute_path.exists():
+        log.error(f"[red]{target} already exists!")
+        exit(1)
+
+    update_source(
+        source_base,
+        target_base,
+        source_relative_path,
+        target_relative_path,
+        source_absolute_path,
+        target_absolute_path,
+    )
+
+    update_references_to_source_recursively(project_root, source_absolute_path, target_absolute_path)
+
+    log.debug(f"[green]Moved from {source} to {target} successfully.")
+
+
+def increment_prefixes(source: Path, target_prefix: str):
+    """Increment the numeric part of prefix of all subsequent directories."""
+    parent_dir = source.parent
+    # first determine the pattern of the prefix by comparing the
+    # increment_prefix with prefix of source
+
+    # if new_prefix is longer than source.name then stop with error
+    if len(target_prefix) > len(source.name):
+        log.error(
+            f"[red]Prefix '{target_prefix}' is longer than '{source.name}'. Prefix needs to be same length as prefix in source."
+        )
+        exit(1)
+
+    # split source into two parts, first with the same length of prefix
+    # second part is the rest
+    source_prefix = source.name[: len(target_prefix)]
+    # source_base = source.name[len(target_prefix) :]
+
+    # determine how many digits are at the end of new_prefix starting from the last character
+    target_prefix_number_of_digits = 0
+    for c in reversed(target_prefix):
+        if c.isdigit():
+            target_prefix_number_of_digits += 1
+        else:
+            break
+    if target_prefix_number_of_digits == 0:
+        log.error(f"[red]Prefix '{target_prefix}' must end with at least one digit.")
+        exit(1)
+
+    # determine how many digits are at the end of source.name starting from the last character
+    source_prefix_number_of_digits = 0
+    for c in reversed(source_prefix):
+        if c.isdigit():
+            source_prefix_number_of_digits += 1
+        else:
+            break
+
+    if source_prefix_number_of_digits == 0:
+        log.error(f"[red]Prefix '{source_prefix}' must end with at least one digit.")
+        exit(1)
+
+    if source_prefix_number_of_digits != target_prefix_number_of_digits:
+        log.error(f"[red]Prefix '{source_prefix}' and '{target_prefix}' must have the same number of digits.")
+        exit(1)
+
+    # get the numeric indeces and calculate the offset
+    source_prefix_index = int(source_prefix[len(source_prefix) - source_prefix_number_of_digits :])
```

Comment:
Python supports indexing from behind

```suggestion
    source_prefix_index = int(source_prefix[-source_prefix_number_of_digits :])
```